src/types/index.ts

```diff
@@ -37,3 +37,26 @@ export interface HtmlInputEvent {
 export interface MetamaskError extends Error {
   code: number | string;
 }
+
+export enum StepState {
+  Todo,
+  Active,
+  WalletOpen,
+  Pending,
+  Success,
+  Warning,
+  Error
+}
+
+export type Step = {
+  tooltip: string;
+  state: StepState;
+};
+
+export type Action = {
+  label: string;
+  loadingLabel: string;
+  pending: boolean;
+  step: Step;
+  promise: () => Promise<void>;
+};
```

Comment:
Since we've moved these up to the global types file I wonder if they should be named more accurately?
```suggestion
export enum TransactionStepState {
  Todo,
  Active,
  WalletOpen,
  Pending,
  Success,
  Warning,
  Error
}

export type TransactionStep = {
  tooltip: string;
  state: StepState;
};

export type TransactionAction = {
  label: string;
  loadingLabel: string;
  pending: boolean;
  step: Step;
  promise: () => Promise<void>;
};
```